lib/policies/userAgentPolicy.ts

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+import { HttpPipelineLogLevel } from "../httpPipelineLogLevel";
+import { HttpRequest } from "../httpRequest";
+import { HttpResponse } from "../httpResponse";
+import { BaseRequestPolicy, RequestPolicy } from "../requestPolicy";
+import { RequestPolicyFactory } from "../requestPolicyFactory";
+import { RequestPolicyOptions } from "../requestPolicyOptions";
+
+/**
+ * Get a RequestPolicyFactory that creates UserAgentRequestPolicies.
+ * @param userAgent The userAgent string to apply to each outgoing request.
+ */
+export function userAgentPolicy(userAgent: string): RequestPolicyFactory {
+    return (nextPolicy: RequestPolicy, options: RequestPolicyOptions) => {
+        return new UserAgentPolicy(userAgent, nextPolicy, options);
+    };
+}
+
+class UserAgentPolicy extends BaseRequestPolicy {
+    constructor(private readonly userAgent: string, nextPolicy: RequestPolicy, options: RequestPolicyOptions) {
+        super(nextPolicy, options);
+    }
+
+    send(request: HttpRequest): Promise<HttpResponse> {
+        if (this.shouldLog(HttpPipelineLogLevel.INFO)) {
+            this.log(HttpPipelineLogLevel.INFO, `Set "User-Agent" header to "${this.userAgent}".`);
+        }
+        request.headers.set("User-Agent", this.userAgent);
```

Comment:
Rikki and I talked about this, and we determined that even if this does get set on the browser, it will essentially be a no-op because the browser will just overwrite it. If we decide that we shouldn't be doing this, even though it's a no-op, then the place to skip it would be when we are constructing the default HttpPipeline, not here in the RequestPolicy.